manifests/init.pp

```diff
@@ -81,6 +84,22 @@
   #Merged values provides by user and default values
   $_real_padoptions = merge($default_padoptions, $padoptions)
 
+  #Install choosing plugins
+  $plugins_list.each |String $_pname, Variant[Boolean, Undef] $_penable| {
```

Comment:
You don't have to type variables here. You already have type checking in the class parameter so IMHO this is redundant.

Reply:
Fixed.